src/components/ControlPanel.vue

```diff
@@ -201,6 +249,55 @@ export default {
         })
     },
 
+    async onAddRemoveAction (data) {
+      this.addRemoveStatus = 'wait' // make them wait for actual action to happen
+      this.addRemoveMode = data.mode
+      this.addRemoveEndDate = this.now
+      this.addRemoveSucceeded = null
+      this.addRemoveFailed = null
+      this.addRemoveWorking = `${data.mode} ${data.status === 'start' ? 'starting…' : 'stopping…'}`
+      const action = data.mode === 'Exclusion' ? `${data.status}Exclusion` : `${data.status}Inclusion`
+      const args = []
+      if (data.isSecure) args.push(data.mode.indexOf('Secure') === 0)
+      this.apiRequest(action, args)
+    },
+
+    onAddRemoveStateChange (controllerStatus) {
```

Comment:
I would move all the function content in the watch method, no need to pass the controllerStatus as it's a computed var